simulation-scripts/perturb.sh

```diff
@@ -356,25 +260,32 @@ run_ssh() {
     root@"${@}"
 }
 
-get_file_to_run() {
-  local FILE="${1}"
-  echo "set -x"
-  if [[ "${IS_DRY_RUN:-}" == 1 ]]; then
-    cat "${FILE}" >&2
-  else
-    cat "${FILE}"
-  fi
-  echo "echo PERTURBANCE COMPLETE FOR ${FILE} ON \$(hostname) AT \$(date)"
-  echo
-}
-
 run_file_on_host() {
   local FILE="${1}"
   local HOST="${2}"
-  (
-    set -x
-    get_file_to_run "${FILE}"
-  ) | run_ssh "${HOST}"
+
+  info "Setup scenario to be perturbed"
+
+  scp \
+    -F "${SSH_CONFIG_FILE}"  \
+    -o "StrictHostKeyChecking=no" \
+    -o "UserKnownHostsFile=/dev/null" \
+    /app/simulation-scripts/"${FILE}" root@"${HOST}":/tmp/setup.sh
+
+  ssh -q -t \
+    -F "${SSH_CONFIG_FILE}" \
+    -o "StrictHostKeyChecking=no" \
+    -o "UserKnownHostsFile=/dev/null" \
+    root@"${HOST}" "chmod +x /root/setup.sh && /tmp/setup.sh"
+
+  info "Cleanup setup"
+  ssh -q -t \
+    -F "${SSH_CONFIG_FILE}" \
+    -o "StrictHostKeyChecking=no" \
+    -o "UserKnownHostsFile=/dev/null" \
+    root@"${HOST}" "rm /tmp/setup.sh"
```

Comment:
Can this be in the same step as `/tmp/setup.sh` run above joined with `&&`?

Reply:
oh yeh,super emabaressing